### ISSUE TITLE
How to perform batch prediction for sequence model?

### ISSUE THREAD

**Issue by ilos-vigil** [amaiya/ktrain#342]
For reference, i follow this [example notebook](https://nbviewer.jupyter.org/github/amaiya/ktrain/blob/master/examples/text/CoNLL2002_Dutch-BiLSTM.ipynb) with different dataset. But i don't know how to predict many sentences at once when all i have is array of string. I could simply loop `predictor.predict()` for each sentence, but it's very slow.

```py
X_test = [
  'What a good weather to visit beach',
  'Is it located near the pool'
]
predictor.predict(X_test)
```

**Comment by amaiya** [comment]
Internally, the  `predict` method calls `predict_on_batch`, but the method itself is not currently coded to accept batches of sentences.  This needs to be changed/fixed.  I have marked this as enhancement, and it will be included in the next version.  Thanks for posting issue.

**Comment by ilos-vigil** [comment]
I made short modification to perform batch prediction, maybe it's useful as reference

```py
def batch_predict(self, sentences):
    lang = TU.detect_lang(sentences)
    nerseq = self.preproc.preprocess(sentences, lang=lang, custom_tokenizer=None)
    if not nerseq.prepare_called:
        nerseq.prepare()
    nerseq.batch_size = len(sentences)
    x_true, _ = nerseq[0]
    lengths = nerseq.get_lengths(0)
    y_pred = self.model.predict(x_true)
    y_labels = self.preproc.p.inverse_transform(y_pred, lengths)
    
    # get data
    try:
        probs = np.max(y_pred, axis=2)
    except:
        probs = y_pred.numpy().tolist() # TODO: remove after confirmation (#316)

    return_datas = []
    for x, y, prob in zip(nerseq.x, y_labels, probs):
        return_data = [(x[i], y[i], prob[i]) for i in range(len(x))]
        return_datas.append(return_data)

    return return_datas
```

This is is how i used the function

```py
y_preds = []
doc_count = len(X_test)

for idx_start in tqdm(range(0, doc_count, batch_size)):
    idx_end = idx_start + batch_size
    if idx_end > doc_count:
        idx_end = doc_count

    y_pred = predictor.batch_predict(predictor, X_test[idx_start:idx_end])
    y_preds.extend(y_pred)
```

**Comment by amaiya** [comment]
@ilos-vigil Many thanks for this, as this was very helpful and saved time.  Instead of implementing a `batch_predict` method, I have used your modifications and incorporated them into `NERPredictor.predict` that now accepts a list of sentences.  The sentences are split into batches (default is 32) and predictions are made on each batch.  You can change/increase the `batch_size` either as an argument to `load_predictor` or setting it directly:
```python
predictor.batch_size = 128
results = predictor.predict(list_of_sentences)
results = predictor.predict(list_of_sentences, return_proba=True)
results = predictor.predict(list_of_sentences, merge_tokens=True)
```

The changes are in the `develop` branch, but have not been tested much.  If you are able to test this on your end, it would be very helpful.  You can install the `develop` branch with:

```
pip3 install git+https://github.com/amaiya/ktrain@develop
```

**Comment by ilos-vigil** [comment]
I tested it with my own code (https://www.kaggle.com/ilosvigil/scl-2021-2-ktrain-bilstm-crf) and it works.

**Comment by amaiya** [comment]
@ilos-vigil This update has been included in v0.26.2 of **ktrain**.  Thanks again for the assistance.
